### PR TITLE
Weigh in

### PR DESCRIPTION
**Pull request by Siffster85**

### CODE REVIEW

File: api.tsx [hunk 1]
```diff
@@ -158,3 +158,39 @@ export const removeUser = async () => {
...
+  try {
+    const user = auth.currentUser;
+    const userAccessToken = await user?.getIdToken(true);
+
+    const response = await instance.post(
+      `/schedules/${date}/plan/weight`,
+      { weight },
+      {
+        headers: {
+          Authorization: `Bearer ${userAccessToken}`,
```
Comment (elenatrr) [inline api.tsx:172]: does it post it? do you have to add content-type header?

File: api.tsx [hunk 2]
```diff
@@ -158,3 +158,39 @@ export const removeUser = async () => {
...
+        },
+      },
+    );
+    return response;
+  } catch (error) {
+    console.log(error);
```
Comment (elenatrr) [inline api.tsx:178]: alert?

Comment (elenatrr) [reply api.tsx:178]: ot throw

File: app/(tabs)/plan/weigh-in-page.tsx [hunk 3]
```diff
@@ -12,12 +14,14 @@ import {
 
 const WeighIn = () => {
   const [weight, setWeight] = useState('');
-
+  const patchWeight = Number(weight);
+  const currentDate = formatDate(new Date());
   async function handleSave() {
     try {
+      const post = await postWeight(currentDate, patchWeight);
+    } catch (error) {
```
Comment (elenatrr) [inline app/(tabs)/plan/weigh-in-page.tsx:22]: i would add error handling here

File: api.tsx (hunk 2 above)
Comment (Karumu05) [inline api.tsx:178]: Could Throw error instead of console.log it for sake of linter

File: app/(tabs)/profile/editProfile.tsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+  const [weight, setWeight] = useState(userProfile.user.weight);
+  const [selectedActivity, setSelectedActivity] = useState('sedentary');
+
+  useEffect(() => {
+    try {
+      const userData = async () => {
+        const data = await getUserProfile();
+        setUserProfile(data.data);
+      };
+      userData();
```
Comment (elenatrr) [inline app/(tabs)/profile/editProfile.tsx:41]: there is no need to put 
```
const data = await getUserProfile();
setUserProfile(data.data);
```
        inside userData() function,

File: app/(tabs)/profile/editProfile.tsx [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    name: userName || userProfile.user.name,
+    dailyActivityLevel: selectedActivity || userProfile.user.dailyActivityLevel,
+    email: email || userProfile.user.email,
+    weight: weight || userProfile.user.weight,
+  };
+
+  async function handleSubmit() {
+    try {
+      const patch = await patchUser(profileData);
+    } catch (error) {
```
Comment (elenatrr) [inline app/(tabs)/profile/editProfile.tsx:57]: error handling?

**Review by elenatrr — CHANGES_REQUESTED** [review]

File: api.tsx (hunk 1 above)
Comment (Siffster85) [reply api.tsx:172]: Not needed and it logs to the db.

File: api.tsx (hunk 2 above)
Comment (Siffster85) [reply api.tsx:178]: I've already fixed that to a throw.

**Pull request merged**